subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/ComponentMetadataHandlerInternal.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.artifacts.dsl;
+
+import org.gradle.internal.component.external.model.VariantDerivationStrategy;
+
+public interface ComponentMetadataHandlerInternal {
```

Comment:
Is it by design to not extend `ComponentMetadataHandler`?

Reply:
Yes, that's just to avoid the actual implementation extending this interface instead of `ComponentMetadataHandler` directly. No big deal either way.